generic/tests/ping.py

```diff
@@ -58,76 +63,153 @@ def _get_loss_ratio(output):
+            resultStr = "packet size 65508 is too large. Maximum is 65507"
+            if output.find(resultStr) == -1:
+                raise error.TestFail(("Ping test with exceeded package failed,"
+                                      " expected string: '%s' not found" % resultStr))
+            #ping with flood
+            logging.info("Flood ping test")
+            utils_test.ping(ip, None, flood=True, output_func=None,
+                            interface=interface, timeout=flood_minutes * 60)
+
+            logging.info("Ping test after flood ping, Check if the network is"
+                         " still alive")
+            status, output = utils_test.ping(ip, counts, interface=interface,
+                                             timeout=float(counts) * 1.5)
             _get_loss_ratio(output)
 
             if status != 0:
-                raise error.TestFail("Ping failed, status: %s,"
-                                     " output: %s" % (status, output))
-
-        error.context("Flood ping test", logging.info)
-        utils_test.ping(ip, None, flood=True, output_func=None,
-                        interface=interface, timeout=flood_minutes * 60)
-
-        error.context("Ping test after flood ping, Check if the network is"
-                      " still alive", logging.info)
-        status, output = utils_test.ping(ip, counts, interface=interface,
-                                         timeout=float(counts) * 1.5)
-        _get_loss_ratio(output)
+                raise error.TestFail("Ping returns non-zero value %s" % output)
 
-        if status != 0:
-            raise error.TestFail("Ping returns non-zero value %s" % output)
-
-        if ping_ext_host:
-            error.base_context("Ping test from guest side,"
-                               " dest: '%s'" % ext_host, logging.info)
-            pkt_sizes = packet_sizes
-            # There is no ping program for guest, so let's hardcode...
-            cmd = ['ping']
-            cmd.append(ext_host)  # external host
-
-            if params.get("os_type") == "windows":
-                cmd.append("-n 10")
-                cmd.append("-l %s")
-                # Windows doesn't support ping with packet
-                # larger than '65500'
-                pkt_sizes = [p for p in packet_sizes if p < 65500]
-                # Add a packet size just equal '65500' for windows
-                pkt_sizes.append(65500)
-            else:
-                cmd.append("-c 10")  # ping 10 times
-                cmd.append("-s %s")  # packet size
-            cmd = " ".join(cmd)
-            for size in pkt_sizes:
-                error.context("Ping with packet size %s" % size,
-                              logging.info)
-                status, output = session.cmd_status_output(cmd % size,
-                                                           timeout=60)
+    if ping_ext_host:
+        logging.info("Ping test from guest side,"
```

Comment:
It is suggested to use error_context.context("xxx", logging.info) for test steps prompts.